repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -0,0 +1,174 @@
+    conflicts("+openmp", when="+hip", msg="Cannot enable both OpenMP and HIP")
+    conflicts("+cuda", when="+sycl", msg="Cannot enable both CUDA and SYCL")
+    conflicts("+hip", when="+sycl", msg="Cannot enable both HIP and SYCL")
+    conflicts("+fft", when="~mpi", msg="FFT support requires MPI")
+    conflicts("+radiation", when="platform=darwin", msg="Radiation is not supported on macOS")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("ERF_ENABLE_MPI", "mpi"),
+            self.define_from_variant("ERF_ENABLE_OMP", "openmp"),
+            self.define_from_variant("ERF_ENABLE_NETCDF", "netcdf"),
+            self.define_from_variant("ERF_ENABLE_PARTICLES", "particles"),
+            self.define_from_variant("ERF_ENABLE_MULTIBLOCK", "multiblock"),
+            self.define_from_variant("ERF_ENABLE_RADIATION", "radiation"),
+            self.define_from_variant("ERF_BUILD_TESTS", "tests"),
+            self.define_from_variant("ERF_BUILD_FCOMPARE", "fcompare"),
+            self.define_from_variant("ERF_ENABLE_FFT", "fft"),
+            self.define("ERF_DIM", "3"),
+        ]
+        args.append(self.define("ERF_USE_EXTERNAL_AMREX", False))
+        args.append(self.define("ERF_CLONE_AMREX", True))
+        args.append(self.define("GIT_SUBMODULE_PROTOCOL", "https"))
+        args.append(self.define("MPIEXEC_PREFLAGS", "--oversubscribe"))
+        args.append(self.define("CMAKE_BUILD_TYPE", "Release"))
+        args.append(self.define("ERF_DIM", "3"))
+
+        if "+netcdf" in self.spec:
+            args.extend(
+                [
+                    self.define("NetCDF_C_PATH", self.spec["netcdf-c"].prefix),
+                    self.define("NetCDF_FORTRAN_PATH", self.spec["netcdf-fortran"].prefix),
+                ]
+            )
+
+        if "+radiation" in self.spec:
+            if "+cuda" in self.spec:
+                args.append(self.define("Kokkos_ARCH_AMPERE80", True))
+            args.extend(
+                [
+                    self.define("ERF_ENABLE_KOKKOS", True),
+                    self.define("Kokkos_ARCH_AMPERE", True),
+                    self.define("ERF_ENABLE_EKAT", True),
+                    self.define("ERF_ENABLE_RRTMGP", True),
+                ]
+            )
+
+        if "+cuda" in self.spec:
+            archs = self.spec.variants["cuda_arch"].value or ["80"]
+            args.append(self.define("CMAKE_CUDA_ARCHITECTURES", ";".join(archs)))
+            args.extend(
+                [
+                    self.define("ERF_ENABLE_CUDA", "ON"),
+                    self.define("CUDAToolkit_ROOT", self.spec["cuda"].prefix),
+                ]
+            )
+
+        if "+hip" in self.spec:
+            args.append(self.define("CMAKE_HIP_ARCHITECTURES", "gfx906;gfx908;gfx90a"))
+
+        print(args)
+        return args
+
+    def setup_build_environment(self, env):
+        super(Erf, self).setup_build_environment(env)
```

Comment:
```suggestion
        super().setup_build_environment(env)
```
Python 3 lets us be a little less verbose with this.